src/llama_stack_client/lib/agents/agent.py

```diff
@@ -19,12 +23,76 @@ def __init__(
         client: LlamaStackClient,
         agent_config: AgentConfig,
         custom_tools: Tuple[CustomTool] = (),
+        memory_bank_id: Optional[str] = None,
     ):
         self.client = client
         self.agent_config = agent_config
         self.agent_id = self._create_agent(agent_config)
         self.custom_tools = {t.get_name(): t for t in custom_tools}
         self.sessions = []
+        self.memory_bank_id = memory_bank_id
+
+    @classmethod
+    def with_memory(
+        cls,
+        client: LlamaStackClient,
+        agent_config: AgentConfig,
+        custom_tools: Tuple[CustomTool] = (),
+        memory_bank_id: Optional[str] = None,
+    ) -> "Agent":
+        if memory_bank_id is None:
+            memory_bank_id = "memory_bank_" + str(uuid.uuid4())
+            client.memory_banks.register(
+                memory_bank_id=memory_bank_id,
+                params={
+                    "memory_bank_type": "vector",
+                    "embedding_model": "all-MiniLM-L6-v2",
+                    "chunk_size_in_tokens": 512,
+                    "overlap_size_in_tokens": 64,
+                },
+            )
+        # find the memory runtime provider
+        providers = client.providers.list()
+        memory_runtime_providers = [
+            p.provider_id
+            for p in providers["tool_runtime"]
+            if p.provider_type == "inline::memory-runtime"
+        ]
+        if len(memory_runtime_providers) == 0:
+            raise ValueError("Memory runtime provider not found")
+        memory_runtime_provider = memory_runtime_providers[0]
+        tool_name = "memory-tool" + str(uuid.uuid4())
+        client.toolgroups.register(
+            tool_group_id="memory_group",
+            tool_group=UserDefinedToolGroupDef(
+                type="user_defined",
+                tools=[
+                    CustomToolDefParam(
+                        type="custom",
```

Comment:
Assume this could also be "user_defined"?

Reply:
yes, this will be updated once we have finalized the types for the server side.

Reply:
updated.